src/Appium.Net/Appium/Service/AppiumServiceBuilder.cs

```diff
@@ -306,6 +307,33 @@ public AppiumServiceBuilder WithStartUpTimeOut(TimeSpan startUpTimeout)
             return this;
         }
 
+        /// <summary>
+        /// Command line arguments that will be passed to NodeJS when it starts up.
+        /// </summary>
+        /// <param name="arguments">
+        /// A collection of arguments that will be passed to NodeJS. Spaces will automatically
+        /// be added between arguments. Arguments cannot be null, empty strings, or only whitespace.
+        /// </param>
+        /// <returns></returns>
+        public AppiumServiceBuilder WithNodeArguments(params string[] arguments)
+        {
+            if (arguments == null)
+            {
+                throw new ArgumentNullException(nameof(arguments), "Argument cannot be null");
+            }
+
+            foreach (var arg in arguments)
+            {
+                if (string.IsNullOrWhiteSpace(arg))
+                {
+                    throw new ArgumentException("Node arguments cannot be null, empty, or only whitespace.", nameof(arguments));
```

Comment:
I would rather show the index of the failing argument here to help users to fix the issue

Reply:
Added